### DISCUSSION TITLE
Implementing the View part for a new framework like Godot or Unity

### DISCUSSION THREAD

**Discussion by JurgenCruz** [reactiveui/ReactiveUI#3641] [Q&A]
Hi, I would like to try out ReactiveUI for a new project which is a videogame. I don't see Godot or Unity packages like I see WPF or Android. I guess I could use the base package for the ViewModel part and implement the binding myself. But I was just wondering how would one go about implementing the framework specific part?

**Comment by anaisbetts** [comment]
Search the codebase for files named "Registrations.cs", you generally want to think about implementing some of those interfaces - you actually can even try it in a sample project directly, you don't need to edit ReactiveUI internals to add support for a new platform, just register those interfaces with Splat on startup similarly to how these files are doing it. 

The most important thing to implement is an IScheduler for MainThreadScheduler - the Android implementation is probably a good example of that but how to implement it will be pretty specifically tied to Unity/Godot. Otherwise, you'll find that a lot of ReactiveUI will Just Work out of the box with new platforms and that everything else is more of a "nice to have" rather than a requirement

**Reply by JurgenCruz** [reply]
Thanks! this is really useful. I will have a look. Glad to hear that some stuff is just nice to have. What about the binding part? I would assume that has to be framework specific.

**Reply by anaisbetts** [reply]
Nope, bindings are not per-platform. The only thing that is per-platform is that if your platform has a unique way to signify when an object has changed - if it does (and that feature is actually commonly used), you can implement `ICreatesObservableForProperty`, https://github.com/reactiveui/ReactiveUI/blob/main/src/ReactiveUI/Platforms/apple-common/KVOObservableForProperty.cs is an example of a "Weird" platform-specific handler

**Comment by dpvreony** [comment]
you're going to get limited mileage in Unity at the moment due to the buildchain. someone did a custom implementation of Rx ([UnityRx](https://github.com/neuecc/UniRx)) which would need work to try and get ReactiveUI to target, there are changes being made to Unity that will give full support for the .NET CLR and thus enable direct use of System.Reactive but these are not due until 2024. see https://blog.unity.com/engine-platform/unity-and-net-whats-next and the thread https://forum.unity.com/threads/unity-future-net-development-status.1092205/page-42

**Reply by JurgenCruz** [reply]
Yeah, I'm aware of UniRx and the limitations of System.Reactive in Unity. This and the whole CEO scandal thing was what was pushing me to Godot in the first place.

**Reply by glennawatson** [reply]
One of our maintainers works at Unity and is working on some of those functionality changes for 2024.
